### DISCUSSION TITLE
Multiple Frame - FrameName class is not defined

### DISCUSSION THREAD

**Discussion by B-Wallace91** [TomSchimansky/CustomTkinter#1277] [Q&A] [answered]
Hello all!

First of all I am having lots of fun using CTk. Take into account that I am a super newbie at Python so apologise for any dumb mistake I made.

I am trying to follow the documentation about "Multiple Frames" where it is recommended to separate the frames in different .py files so that the code is lighter and easier to read (I did the same for the TopLevelWindow). I have several .py files, of which one is the main App:

```python
class App(customtkinter.CTk):
    def __init__(self):
        super().__init__()
        self.geometry("900x600")
        self.title("Perma_Chooser2.0")
        # set geometry
        self.grid_columnconfigure(1, weight=1)
        self.grid_columnconfigure((2, 3), weight=0)
        self.grid_rowconfigure((0, 1, 2), weight=0)
        # selection.py frame = where user writes permalink, chooses type and locale and submits values
        self.perma_selection = SelectionFrame(self, header_name="Permalink Chooser")
        self.perma_selection.grid(row=0, column=0, padx=20, pady=20)
if __name__ == "__main__":
    app = App()
    app.mainloop()
```

I have created a separate .py file to write the SelectionFrame CTkFrame and it starts like this:

```python
class SelectionFrame(customtkinter.CTkFrame):
    def __init__(self, *args, header_name="Permalink Chooser", **kwargs):
        super().__init__(*args, **kwargs)
        self.header_name = header_name
        self.header = customtkinter.CTkLabel(self, text=self.header_name)
        self.header.grid(row=0, column=0, padx=10, pady=10)
blablabla
```

However, when I run my main py file I get the following error:
```
line 17, in __init__
    self.perma_selection = SelectionFrame(self, header_name="Permalink Chooser")
                           ^^^^^^^^^^^^^^
NameError: name 'SelectionFrame' is not defined
```

Now, I am sure I am missing something stupid, but I just don't understand what that is. Could you please help me?
Thank you a lot for taking a look!

**Comment by Wolf-SO** [comment, marked as answer]
You have to import the file that defines the separated class (say `selframe.py`) like this:

```python
import selframe
```

... in the main file before you using it.

**Reply by B-Wallace91** [reply]
Hello Wolf-SO!

Thank you for your help! As a matter of fact it was a very rookie mistake I made!

I should have simply wrote:

```python
from name_of_file import name_of_class
```

Thank you again!

**Reply by Wolf-SO** [reply]
Great to see it helped. 😄
